File: Daily_Challenge/20230320_LC_2348.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        i, n = 0, len(nums)
        ans = 0
        while i < n:
            if nums[i] == 0:
                j = i + 1
                while j < n and nums[j] == 0:
                    j += 1
                l = j - i + 1
                ans += l * (l-1) // 2
                i = j
            i += 1
        return ans

File: Daily_Challenge/20230320_LC_2348.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        long long ans = 0;
        int i = 0, n = nums.size();

        while (i < n){
            if (nums[i] == 0){
                int j = i + 1;
                while(j < n && nums[j] == 0){
                    j++;
                }
                long long l = j - i + 1;
                ans += (l * (l-1))/2;
                i = j;
            }
            i++;
        }
        return ans;
    }
};